source/agora/common/Deserializer.d

```diff
@@ -149,6 +150,17 @@ public T deserializeFull (T) (scope ubyte[] data) @safe
     return value;
 }
 
+/// Ditto
+public void deserializePart (T) (ref T record, scope DeserializeDg dg) @safe
+    if (isInputRange!T && hasLength!T)
```

Comment:
Does `hasLength` check if `.length` is writable though?

Reply:
I guess it doesn't matter much, we get a compiler error either way.

Reply:
Don't think so